### PR TITLE
docs: Added clarification about UART timeouts

### PR DESCRIPTION
**Pull request by lorenzschmid**
As discussed in #2660 the two UART timeouts given to the constructor (first char, between chars) is valid for reading **and** writing. Updated the documentation to clarify.

### CODE REVIEW

File: docs/library/pyb.UART.rst [hunk 1]
```diff
@@ -87,7 +87,7 @@ Methods
          - ``stop`` is the number of stop bits, 1 or 2.
          - ``flow`` sets the flow control type. Can be 0, ``UART.RTS``, ``UART.CTS``
            or ``UART.RTS | UART.CTS``.
-         - ``timeout`` is the timeout in milliseconds to wait for the first character.
+         - ``timeout`` is the timeout in milliseconds to wait for writing/reading the first character.
          - ``timeout_char`` is the timeout in milliseconds to wait between characters.
```
Comment (dpgeorge) [inline docs/library/pyb.UART.rst]: Do you think this line should also be clarified?  Eg at the end add ", for both reading and writing."

File: docs/library/pyb.UART.rst [hunk 2]
```diff
@@ -111,11 +111,6 @@ Methods
 
        Returns the number of bytes waiting (may be 0).
 
-    .. method:: UART.writechar(char)
-
-      Write a single character on the bus.  ``char`` is an integer to write.
-      Return value: ``None``. See note below if CTS flow control is used.
-
```
Comment (dpgeorge) [inline docs/library/pyb.UART.rst]: Reordering the description for this method is a separate thing, nothing to do with timeout, so please put it in a separate commit (but same PR if you like).

File: docs/library/pyb.UART.rst (hunk 1 above)
Comment (lorenzschmid) [inline docs/library/pyb.UART.rst]: True, added them in a new PR!

File: docs/library/pyb.UART.rst (hunk 2 above)
Comment (lorenzschmid) [inline docs/library/pyb.UART.rst]: Thanks for the hint, corrected it in a new PR

**Comment by lorenzschmid** [comment]
Created new PR based on the suggestions here #2686.
